### PR TITLE
WIP: Make create_store return a class

### LINKED ISSUE #4: Make store a class instance rather than dict?
So you can:

``` python
store.subscribe(listener)
```

rather than:

``` python
store['subscribe'](listener)
```

### PR DESCRIPTION
**Pull request by kasbah**
This is a start to fix #4. I am porting tests from https://github.com/reactjs/redux/blob/master/test/createStore.spec.js to go along with this to make sure it still works as we expect. 

Any thoughts so far?

### CODE REVIEW

File: pydux/create_store.py [hunk 1]
```diff
@@ -81,17 +73,17 @@ def dispatch(action):
...
 
-        if is_dispatching[0]:
+        if self.is_dispatching[0]:
             raise Exception('Reducers may not dispatch actions.')
 
         try:
-            is_dispatching[0] = True
-            current_state[0] = reducer(current_state[0], action)
+            self.is_dispatching[0] = True
+            self.current_state[0] = self.current_reducer[0](self.current_state[0], action)
```
Comment (kasbah) [inline pydux/create_store.py:81]: Not sure if this was originally a bug. To me it looks like it wouldn't dispatch properly when the reducer was replaced the way it is on master.

File: pydux/create_store.py [hunk 2]
```diff
@@ -16,63 +16,55 @@ class ActionTypes(object):
...
+            return enhancer(create_store)(reducer)
+
+        if not hasattr(reducer, '__call__'):
+            raise TypeError('Expected the reducer to be a function.')
+
+        self.reducer = reducer
+        self.initial_state = initial_state
+        self.enhancer = enhancer
+        # single-element arrays for r/w closure
+        self.current_reducer = [reducer]
```
Comment (usrlocalben) [inline pydux/create_store.py:33]: All of these list[] wrappers and [0] indexes are for allowing read/write access to variables in closure.  If it is replaced with a class instance, than they can all go away.

File: pydux/create_store.py (hunk 1 above)
Comment (usrlocalben) [reply pydux/create_store.py:81]: it looks to me like you are correct

**Comment by kasbah** [comment]
Let's close this, current implementation seems fine.

**Pull request closed without merging**